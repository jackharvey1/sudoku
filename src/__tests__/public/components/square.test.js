import React from 'react';
import Square from '../../../public/components/square';
import ShallowRenderer from 'react-test-renderer/shallow';

let renderer;

beforeEach(() => {
    renderer = new ShallowRenderer();
});

it('renders as expected in basic state', () => {
    renderer.render(<Square
        value={1}
        onClick={() => {}}
        box={0}
        square={0}
        isSelected={false}
        isRelevant={false}
        isLocked={false}
    />);
    const output = renderer.getRenderOutput();
    expect(output).toMatchSnapshot();
});

it('renders with bold text when locked', () => {
    renderer.render(<Square
        value={1}
        onClick={() => {}}
        box={0}
        square={0}
        isSelected={false}
        isRelevant={false}
        isLocked={true}
    />);
    const output = renderer.getRenderOutput();
    expect(output).toMatchSnapshot();
});

it('renders with background when selected', () => {
    renderer.render(<Square
        value={1}
        onClick={() => {}}
        box={0}
        square={0}
        isSelected={true}
        isRelevant={false}
        isLocked={false}
    />);
    const output = renderer.getRenderOutput();
    expect(output).toMatchSnapshot();
});

it('renders with a light background when on the same row', () => {
    renderer.render(<Square
        value={1}
        onClick={() => {}}
        box={0}
        square={0}
        isSelected={true}
        isRelevant={true}
        isLocked={false}
    />);
    const output = renderer.getRenderOutput();
    expect(output).toMatchSnapshot();
});
