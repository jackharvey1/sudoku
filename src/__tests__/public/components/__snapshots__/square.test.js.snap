// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected in basic state 1`] = `
<div
  className="css-iz9zke"
  onClick={[Function]}
>
  <p
    className="css-1h92lle"
  >
    1
  </p>
</div>
`;

exports[`renders with a light background when on the same row 1`] = `
<div
  className="css-9v78r0"
  onClick={[Function]}
>
  <p
    className="css-1h92lle"
  >
    1
  </p>
</div>
`;

exports[`renders with background when selected 1`] = `
<div
  className="css-9v78r0"
  onClick={[Function]}
>
  <p
    className="css-1h92lle"
  >
    1
  </p>
</div>
`;

exports[`renders with bold text when locked 1`] = `
<div
  className="css-14zxudu"
  onClick={[Function]}
>
  <p
    className="css-1h92lle"
  >
    1
  </p>
</div>
`;
